File: Core/Workspace/CSharp/Adapters/Allors.Workspace.Adapters.Local/Workspace/Workspace.cs
// <copyright file="LocalWorkspace.cs" company="Allors bvba">
// Copyright (c) Allors bvba. All rights reserved.
// Licensed under the LGPL license. See LICENSE file in the project root for full license information.
// </copyright>

namespace Allors.Workspace.Adapters.Local
{
    public class Workspace : Adapters.Workspace
    {
        public Workspace(DatabaseConnection database) : base(database) => this.Lifecycle.OnInit(this);

        public long UserId => ((DatabaseConnection)this.Database).UserId;

        public override ISession CreateSession() => new Session(this, this.Lifecycle.CreateSessionContext());
    }
}

File: typescript/libs/system/workspace/adapters-json/src/lib/workspace/workspace.ts
import { Workspace as SystemWorkspace } from '@allors/system/workspace/adapters';
import { ISession } from '@allors/system/workspace/domain';
import { DatabaseConnection } from '../database/database-connection';
import { Session } from '../session/session';

export class Workspace extends SystemWorkspace {
  constructor(database: DatabaseConnection) {
    super(database);
  }

  createSession(): ISession {
    return new Session(this);
  }
}
